### PR TITLE
UNR-4405 - Suppress welcome message

### PR DESCRIPTION
**Pull request by jayprobable**
Override ShouldBroadCastWelcomeMessage to suppress sending of welcome messages when simulated players connect.

So at this time we've decided to supress the welcome message, given that it spamming to hundreds of players it seems reasonable that for games with high population this would not be desired.

Testing:
https://buildkite.com/improbable/unrealgdk-nfr/builds/3051

The UNR-4405 Jira ticket has examples of the log spam it generates.

Chose to create ABenchmarkGymPlayerState because it "felt" like the cleanest choice to override the existing virtual function.
It may be useful in the future to customize PlayerState for the benchmarking scenarios as well.

### CODE REVIEW

File: Game/Source/GDKTestGyms/BenchmarkGymPlayerState.cpp [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#include "BenchmarkGymPlayerState.h"
+
+ABenchmarkGymPlayerState::ABenchmarkGymPlayerState()
+{
+
+}
+
+bool ABenchmarkGymPlayerState::ShouldBroadCastWelcomeMessage(bool bExiting)
+{
+	//we do not want to send notifications between players
```
Comment (m-samiec) [inline Game/Source/GDKTestGyms/BenchmarkGymPlayerState.cpp:10]:
```suggestion
	// We do not want to send notifications between players.
```

File: Game/Source/GDKTestGyms/BenchmarkGymPlayerState.cpp [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#include "BenchmarkGymPlayerState.h"
+
+ABenchmarkGymPlayerState::ABenchmarkGymPlayerState()
+{
+
+}
+
+bool ABenchmarkGymPlayerState::ShouldBroadCastWelcomeMessage(bool bExiting)
```
Comment (m-samiec) [inline Game/Source/GDKTestGyms/BenchmarkGymPlayerState.cpp:8]:
```suggestion
bool ABenchmarkGymPlayerState::ShouldBroadCastWelcomeMessage(bool /*bExiting*/)
```

File: Game/Source/GDKTestGyms/BenchmarkGymPlayerState.cpp [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#include "BenchmarkGymPlayerState.h"
+
+ABenchmarkGymPlayerState::ABenchmarkGymPlayerState()
```
Comment (MatthewSandfordImprobable) [inline Game/Source/GDKTestGyms/BenchmarkGymPlayerState.cpp:3]: Don't think we need this constructor.

File: Game/Source/GDKTestGyms/BenchmarkGymPlayerState.cpp (hunk 2 above)
Comment (MatthewSandfordImprobable) [reply Game/Source/GDKTestGyms/BenchmarkGymPlayerState.cpp:8]: bool bExiting /*=  false*/

**Review by m-samiec — COMMENTED** [review]
Was there an NFR run against this branch? Did it supress the ring buffer spam?

File: Game/Source/GDKTestGyms/BenchmarkGymPlayerState.cpp (hunk 2 above)
Comment (MatthewSandfordImprobable) [reply Game/Source/GDKTestGyms/BenchmarkGymPlayerState.cpp:8]: Is there a nice reason to comment out the arg if it is not used?

Comment (m-samiec) [reply Game/Source/GDKTestGyms/BenchmarkGymPlayerState.cpp:8]: Some compilers will warn if the variable isn't used

Comment (jayprobable) [reply Game/Source/GDKTestGyms/BenchmarkGymPlayerState.cpp]: done

File: Game/Source/GDKTestGyms/BenchmarkGymPlayerState.cpp (hunk 1 above)
Comment (jayprobable) [reply Game/Source/GDKTestGyms/BenchmarkGymPlayerState.cpp]: done

File: Game/Source/GDKTestGyms/BenchmarkGymPlayerState.cpp (hunk 3 above)
Comment (jayprobable) [reply Game/Source/GDKTestGyms/BenchmarkGymPlayerState.cpp]: done

**Comment by jayprobable** [comment]
> Was there an NFR run against this branch? Did it supress the ring buffer spam?

This [NFR run](https://buildkite.com/improbable/product-research-benchmarks-run/builds/16996) no longer has the spam from the non-authoritative server calling a client RPC, additionally spot checking these [logs](https://buildkite.com/organizations/improbable/pipelines/product-research-benchmarks-run/builds/16996/jobs/7c606dd6-d2d0-426e-9b0d-1b66db4f080d/artifacts/5697be2f-5a3d-4f79-a5bb-2511f36ff653) showed no instances of rpc ring buffer being full.

**Review by yujunting — APPROVED** [review]

**Review by siliviali — APPROVED** [review]

**Pull request merged**